### PR TITLE
PB-428 send participant state

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### References

[PB-429 send participant state](https://xainag.atlassian.net/browse/PB-429)

### Summary

* added current state and current round in `HeartbeatRequest`

### Are there any open tasks/blockers for the ticket?

[PB-431](https://xainag.atlassian.net/browse/PB-431)

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [x] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [x] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [x] Linked the ticket in the merge request title or the references section.
- [x] Added an informative merge request summary.

**Code checklist**

- [x] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [x] Passed scope checks.
- [x] Added or updated tests if needed.
- [x] Added or updated code documentation if needed.
- [x] Conforms to Google docstring style.
- [x] Conforms to XAIN structlog style.

### CODE REVIEW

File: xain_sdk/participant_state_machine.py [hunk 1]
```diff
@@ -304,7 +304,9 @@ def message_loop(
 
     coordinator: CoordinatorStub = CoordinatorStub(channel=channel)
     while not terminate.is_set():
-        request = HeartbeatRequest()
+        request = HeartbeatRequest(
+            state=par_state_to_proto_state(state_record.state), round=state_record.round
```
Comment (finiteprods) [inline xain_sdk/participant_state_machine.py]: better to use
```python
state, round = state_record.lookup()
```
to be on the safe side

File: xain_sdk/participant_state_machine.py [hunk 2]
```diff
@@ -407,3 +409,22 @@ def begin_training(
...
+    Returns:
+        The corresponding ``xain_proto.fl.coordinator_pb2.State``.
+    """
+    if state == ParState.TRAINING:
+        return State.TRAINING
+
+    if state == ParState.WAITING:
+        return State.READY
+
+    return State.FINISHED
```
Comment (finiteprods) [inline xain_sdk/participant_state_machine.py]: again on the safe side, you can check that
```python
if state == ParState.DONE:
```
(and log if it's none of the above)

Comment (Robert-Steiner) [reply xain_sdk/participant_state_machine.py]: Thanks for the review👍 I applied your requested changes.

File: xain_sdk/participant_state_machine.py [hunk 3]
```diff
@@ -407,3 +408,32 @@ def begin_training(
...
+    Args:
+        state: The ``ParState``.
+
+    Returns:
+        The corresponding ``xain_proto.fl.coordinator_pb2.State``.
+
+
+    Raises:
+
+        ValueError: If the ``ParState`` can not be converted into a
```
Comment (finiteprods) [inline xain_sdk/participant_state_machine.py]:
```suggestion
        ValueError: If the ``ParState`` cannot be converted into a
```

**Pull request merged**